cookbook/ifan02.rst

```diff
@@ -0,0 +1,181 @@
+        id: vbutton_relay_1
+        pin:
+          number: GPIO9
+          inverted: True
+        on_press:
+          then:
+            - switch.toggle: fan_relay1
+            - switch.turn_on: update_fan_speed
+
+      - platform: gpio
+        id: vbutton_relay_2
+        pin:
+          number: GPIO10
+          inverted: True
+        on_press:
+          then:
+            - switch.toggle: fan_relay2
+            - switch.turn_on: update_fan_speed
+
+      - platform: gpio
+        id: vbutton_relay_3
+        pin:
+          number: GPIO14
+          inverted: True
+        on_press:
+          then:
+            - switch.toggle: fan_relay3
+            - switch.turn_on: update_fan_speed
+
+    output:
+      - platform: custom
+        type: float
+        outputs:
+          id: fanoutput
+        lambda: |-
+          auto ifan02 = new IFan02Output();
+          App.register_component(ifan02);
+          return {ifan02};
+
+      - platform: gpio
+        pin: GPIO12
+        id: light_output
+
+    light:
+      - platform: binary
+        name: ifan02_light
+        output: light_output
+        id: light
+
+    switch:
+      - platform: template
+        id: update_fan_speed
+        optimistic: True
+        turn_on_action:
+          then:
+            - delay: 200ms
+            - if:
+                condition:
+                  and:
+                    - switch.is_off: fan_relay1
+                    - switch.is_off: fan_relay2
+                    - switch.is_off: fan_relay3
+                then:
+                  - fan.turn_off: ifan02
+            - if:
+                condition:
+                  and:
+                    - switch.is_on: fan_relay1
+                    - switch.is_off: fan_relay2
+                    - switch.is_off: fan_relay3
+                then:
+                  - fan.turn_on:
+                      id: ifan02
+                      speed: LOW
+            - if:
+                condition:
+                  and:
+                    - switch.is_on: fan_relay1
+                    - switch.is_on: fan_relay2
+                    - switch.is_off: fan_relay3
+                then:
+                  - fan.turn_on:
+                      id: ifan02
+                      speed: MEDIUM
+            - if:
+                condition:
+                  and:
+                    - switch.is_on: fan_relay1
+                    - switch.is_off: fan_relay2
+                    - switch.is_on: fan_relay3
+                then:
+                  - fan.turn_on:
+                      id: ifan02
+                      speed: HIGH
+            - switch.turn_off: update_fan_speed
+
+      - platform: gpio
+        pin: GPIO5
+        id: fan_relay1
+
+      - platform: gpio
+        pin: GPIO4
+        id: fan_relay2
+
+      - platform: gpio
+        pin: GPIO15
+        id: fan_relay3
+
+    fan:
+      - platform: speed
+        output: fanoutput
+        id: ifan02
+        name: ifan02_fan
```

Comment:
A see also section is missing here. It should point to some of the components used here (fan, custom output, etc)